tests/Integration/Persistence/FactoryWithHooksInInitializeTest.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Zenstruck\Foundry\Tests\Integration\Persistence;
+
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Zenstruck\Foundry\Test\Factories;
+use Zenstruck\Foundry\Tests\Fixture\Factories\WithHooksInInitializeFactory;
+
+/**
+ * @author Nicolas Philippe <nikophil@gmail.com>
+ */
+final class FactoryWithHooksInInitializeTest extends KernelTestCase
```

Comment:
I did not find any other class to put this test (we need a `KernelTestCase`, so that `->isPersisting()` can return true)